### PR TITLE
New folder in copy/move dialog, closes #2093

### LINKED ISSUE #2093: Create new folders when moving or copying
A pretty standard feature of file managers is to be able to create a folder when copying or moving files. For example, Google's own files app and open file manager

This makes organizing / triaging files much easier.

In order to do this we should add a button + dialog to create folders as part of the copy / move dialog.

### PR DESCRIPTION
**Pull request by harshad1**
In this PR I address #2093 

I have added a new file dialog to the move / copy / create dialogs

I have also reworked the UI to use floating buttons

### CODE REVIEW

**Comment by harshad1** [comment]
This works. But I am having some problems with the search bar.

1. The keyboard does not show properly on RequestFocus
2. The keyboard is partially hidden by the navigation bar


These problems also exist in the current move dialog but should be fixed before I am ready to call this done.

**Comment by harshad1** [comment]
image

New dialog

**Comment by harshad1** [comment]
Issue with keyboard

image

File: app/src/main/java/net/gsantner/markor/frontend/filebrowser/MarkorFileBrowserFactory.java [hunk 1]
```diff
@@ -40,10 +40,11 @@ public static GsFileBrowserOptions.Options prepareFsViewerOpts(Context context,
         opts.doSelectFolder = doSelectFolder;
         opts.doSelectFile = !doSelectFolder;
 
-        opts.searchHint = R.string.search_documents;
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/frontend/filebrowser/MarkorFileBrowserFactory.java:43]: Not the best default in many cases

File: app/src/main/java/net/gsantner/markor/frontend/textview/TextViewUtils.java [hunk 2]
```diff
@@ -693,13 +693,6 @@ public static boolean checkRange(final int length, final int... indices) {
         return indices != null && indices.length >= 2 && inRange(0, length, indices) && indices[1] > indices[0];
     }
 
-    public static void blinkView(final View view) {
-        if (view != null) {
-            final float init = view.getAlpha();
-            ObjectAnimator.ofFloat(view, View.ALPHA, init, 0.1f, 1.0f, 0.1f, 1.0f, init).setDuration(1000).start();
-        }
-    }
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/frontend/textview/TextViewUtils.java:701]: Moved this to GsContextUtils. Probably should create GsViewUtils or something.

File: app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserFragment.java [hunk 3]
```diff
@@ -642,24 +652,7 @@ private void importFileToCurrentDirectory(Context context, File sourceFile) {
     @Override
     public void setUserVisibleHint(boolean isVisibleToUser) {
         super.setUserVisibleHint(isVisibleToUser);
-        if (isVisibleToUser && getCurrentFolder() != null && !TextUtils.isEmpty(getCurrentFolder().getName()) && getToolbar() != null) {
-            getToolbar().setTitle(getCurrentFolder().getName());
-            reloadCurrentFolder();
-        }
-    }
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserFragment.java:649]: Moved depreciated function to onResume()

File: app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserFragment.java [hunk 4]
```diff
@@ -642,24 +652,7 @@ private void importFileToCurrentDirectory(Context context, File sourceFile) {
...
 
-    private void showAndBlink(final File file) {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserFragment.java:651]: Moved to adapter so I could call it from the dialog version too

File: app/src/main/res/layout/opoc_filesystem_dialog.xml [hunk 5]
```diff
@@ -9,97 +9,101 @@
  *
 #########################################################*/
 -->
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (harshad1) [inline app/src/main/res/layout/opoc_filesystem_dialog.xml:12]: Rewrite of UI..

**Comment by harshad1** [comment]
This seems to be working well. Marking it ready.

File: app/src/main/java/net/gsantner/markor/frontend/textview/TextViewUtils.java (hunk 2 above)
Comment (gsantner) [reply app/src/main/java/net/gsantner/markor/frontend/textview/TextViewUtils.java:701]: It's OK there 😄

**Comment by gsantner** [comment]
@harshad1 
My opinion: I would rather add a bottom-bar similar to editor textactions bar here, the buttons are shadowing potential important information from the listview, as well a bit bloating.

But I still merge this now, want to make sure everything gets a bit together now 😄 . Moving the UI around is rather something quick done any time.

**Comment by gsantner** [comment]
Thank you

**Pull request merged**